src/internal/m365/exchange/restore_test.go

```diff
@@ -369,3 +373,71 @@ func (suite *RestoreIntgSuite) TestRestoreExchangeObject() {
 		})
 	}
 }
+
+func (suite *RestoreIntgSuite) TestRestoreAndBackupEventWithRecurringAndAttachments() {
+	t := suite.T()
+
+	ctx, flush := tester.NewContext(t)
+	defer flush()
+
+	var (
+		userID  = tester.M365UserID(t)
+		subject = testdata.DefaultRestoreConfig("event").Location
+		handler = newEventRestoreHandler(suite.ac)
+	)
+
+	calendar, err := handler.ac.CreateContainer(ctx, userID, subject, "")
+	require.NoError(t, err, clues.ToCore(err))
+
+	calendarID := ptr.Val(calendar.GetId())
+
+	defer func() {
+		// Removes calendar containing events created during the test
+		err = suite.ac.Events().DeleteContainer(ctx, userID, calendarID)
+		assert.NoError(t, err, clues.ToCore(err))
+	}()
+
+	bytes := exchMock.EventWithRecurrenceAndExceptionAndAttachmentBytes("Reoccurring event restore and backup test")
+	info, err := handler.restore(
+		ctx,
+		bytes,
+		userID, calendarID,
+		fault.New(true))
+	assert.NoError(t, err, clues.ToCore(err))
```

Comment:
nit: general rule of thumb is to use `assert` when checking data for the test but the test can continue otherwise (i.e. comparing expected values). Use `require` when checking for things that should halt the test (i.e. failed to get response from remote server and now there's no data to check for expected values)
```suggestion
	require.NoError(t, err, clues.ToCore(err))
```